composer/datasets/cifar10.py

```diff
@@ -36,6 +40,54 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHpara
                 memory_format=self.synthetic_memory_format,
             )
 
+        elif self.use_ffcv:
+            try:
+                from ffcv.fields.decoders import IntDecoder, SimpleRGBImageDecoder
+                from ffcv.loader import Loader, OrderOption
+                from ffcv.transforms import (Convert, Cutout, RandomHorizontalFlip, RandomResizedCrop, RandomTranslate,
+                                             ToDevice, ToTensor, ToTorchImage)
+                from ffcv.transforms.common import Squeeze
+            except ImportError:
+                raise ImportError(
+                    textwrap.dedent("""\
+                    Composer was installed without ffcv support.
+                    To use ffcv with Composer, run `pip install mosaicml[ffcv]`"""))
+
+            if self.datadir is None:
+                raise ValueError("datadir is required if use_synthetic is False")
+
+            cifar10_ffcv_mean = [125.307, 122.961, 113.8575]
+            cifar10_ffcv_std = [51.5865, 50.847, 51.255]
+            label_pipeline: List[Operation] = [IntDecoder(), ToTensor(), Squeeze()]
+            image_pipeline: List[Operation] = [SimpleRGBImageDecoder()]
+
+            if self.is_train:
+                image_pipeline.extend([
+                    RandomHorizontalFlip(),
+                    #RandomTranslate(padding=2, fill=tuple(map(int, cifar10_ffcv_mean))),
+                    #Cutout(4, tuple(map(int, cifar10_ffcv_mean))),
+                ])
+            # Common transforms for train and test
+            image_pipeline.extend([
+                ToTensor(),
+                ToTorchImage(channels_last=False, convert_back_int16=False),
+                Convert(torch.float32),
+                transforms.Normalize(cifar10_ffcv_mean, cifar10_ffcv_std),
+            ])
+
+            ordering = OrderOption.RANDOM if self.is_train else OrderOption.SEQUENTIAL
+
+            filepath = "/cifar10_train.beton" if self.is_train else "/cifar10_test.beton"
+
+            return Loader(self.datadir + filepath,
```

Comment:
this is more future-thinking, but some of our algorithms run via inserting into dataset transforms. Is there an analogus mechnism here where FFCV-compliant transforms can be inserted to the pipelines _after_ the dataloader is created?

Reply:
We can insert any torchvision transforms here with ffcv (or write our own in ffcv format) but we need to add "recompile" flag so that ffcv can recompile the data processing pipeline.

Reply:
Since they are compiling the code to machine code, if we modify in-place we still need to recompile it otherwise it uses stale code.